dbt/adapters/bigquery/connections.py

```diff
@@ -713,8 +714,24 @@ def _query_and_results(
                 self._bq_job_link(query_job.location, query_job.project, query_job.job_id)
             )
 
-        iterator = query_job.result(max_results=limit, timeout=job_execution_timeout)
+        # only use async logic if user specifies a timeout
+        if job_execution_timeout:
+            loop = asyncio.new_event_loop()
+            future_iterator = asyncio.wait_for(
+                loop.run_in_executor(None, query_job.result), timeout=job_execution_timeout
+            )
 
+            try:
+                iterator = loop.run_until_complete(future_iterator)
+            except asyncio.TimeoutError:
+                query_job.cancel()
+                raise DbtDatabaseError(
```

Comment:
Is this necessarily a `DbtDatabaseError`? There are several reasons this could time out, including user error (e.g. cartesian join) and server side error. It might be right, I'm asking for confirmation; I don't actually know what `DbtDatabaseError` covers.

Reply:
yea I'm not sure either, this could also probably be it's own exception eventually `DbtModelTimeout` or something